internal/pkg/agent/application/actions/handlers/handler_action_diagnostics.go

```diff
@@ -241,8 +243,43 @@ func (h *Diagnostics) diagUnits(ctx context.Context) []client.DiagnosticUnitResu
 	return uDiag
 }
 
+// diagUnits gathers diagnostics from components.
+func (h *Diagnostics) diagComponents(ctx context.Context) []client.DiagnosticComponentResult {
+	cDiag := make([]client.DiagnosticComponentResult, 0)
+	h.log.Debug("Performing component diagnostics")
+	startTime := time.Now()
+	defer func() {
+		h.log.Debugf("Component diagnostics complete. Took: %s", time.Since(startTime))
+	}()
+	rr := h.diagProvider.PerformDiagnostics(ctx)
```

Comment:
With this addition we are performing the diagnostics on the components/units twice (one here and one in the diagUnits).

For both performance and consistency reason it would be better to call `PerformDiagnostics` outside of the diagComponents/diagUnits and pass the result as a parameter to both functions

Reply:
Oh you're right, will fix that!